### PR TITLE
plugins/solidigm: Added Telemetry NLog parser support.

### PR DESCRIPTION
**Pull request by lgdacunh**

### CODE REVIEW

File: plugins/solidigm/solidigm-nvme.h [hunk 1]
```diff
@@ -1,6 +1,6 @@
 // SPDX-License-Identifier: GPL-2.0-or-later
 /*
- * Copyright (c) 2022-2023 Solidigm.
+ * Copyright (c) 2022 Solidigm.
```
Comment (igaw) [inline plugins/solidigm/solidigm-nvme.h:3]: Do you really want to drop 2023?

File: plugins/solidigm/solidigm-telemetry/config.c [hunk 2]
```diff
@@ -28,17 +32,46 @@ static bool config_get_by_version(const struct json_object *obj, int version_maj
...
-	if (!json_object_object_get_ex(obj, str_key, &token_obj))
+	if (!json_object_object_get_ex(config, str_key, &token))
 		return false;
-	if  (!config_get_by_version(token_obj, version_major, version_minor, value))
+	if  (!config_get_by_version(token, version_major, version_minor, value))
 		return false;
 	return value != NULL;
 }
+
+
```
Comment (igaw) [inline plugins/solidigm/solidigm-telemetry/config.c:50]: Empty line too much

File: plugins/solidigm/solidigm-telemetry/data-area.c [hunk 3]
```diff
@@ -302,6 +303,27 @@ static int telemetry_log_data_area_get_offset(const struct telemetry_log *tl,
 	return 0;
 }
 
+static int telemetry_log_nlog_parse(const struct telemetry_log *tl, struct json_object *formats,
+			     uint64_t nlog_file_offset,	uint64_t nlog_size,
```
Comment (igaw) [inline plugins/solidigm/solidigm-telemetry/data-area.c:307]: too many empty spaces

**Review by igaw — COMMENTED** [review]
These submission contains white space cleanups and a new feature. Generally, I really would like to see more review friendly submissions. 

Also the function names are really long.

```shell
~/work/linux/scripts/checkpatch.pl -g HEAD 
WARNING: Missing commit description - Add an appropriate one

WARNING: added, moved or deleted file(s), does MAINTAINERS need updating?
#407: 
new file mode 100644

WARNING: quoted string split across lines
#535: FILE: plugins/solidigm/solidigm-telemetry/nlog.c:124:
+               SOLIDIGM_LOG_WARNING("%s:%d with %d header mismatches ( %s). "
+                                    "Configuration file may be missing format headers.",

ERROR: Missing Signed-off-by: line(s)

total: 1 errors, 3 warnings, 484 lines checked

NOTE: For some of the reported defects, checkpatch may be able to
      mechanically convert to the typical style using --fix or --fix-inplace.

Commit 1c5eeb6a8d54 ("plugins/solidigm: Added Telemetry NLog parser support.") has style problems, please review.

NOTE: If any of the errors are false positives, please report
      them to the maintainer, see CHECKPATCH in MAINTAINERS.
```

File: plugins/solidigm/solidigm-nvme.h (hunk 1 above)
Comment (lgdacunh) [reply plugins/solidigm/solidigm-nvme.h:3]: Oops! I am rolling back this change.

File: plugins/solidigm/solidigm-telemetry/data-area.c (hunk 3 above)
Comment (lgdacunh) [reply plugins/solidigm/solidigm-telemetry/data-area.c:307]: I am assuming you meant to request to align argument to the parenthesis

**Comment by lgdacunh** [comment]
@igaw, what looks like a clean-up is actually a change on the first letter of some JSON key strings followed by fixing  checkpatch WARNINGS on the same line.
I signed the commit, added some more description to the commit message, removed the empty line, and aligned the arguments to the function parenthesis.
Part of the reason the function names are long is because I opted for prefixing the global symbols with solidigm_. 
This way I make sure it is unlikely I will have a symbol name conflict with other areas of nvme-cli.
Even keeping the "solidigm_" prefix there is room to abbreviate some words in the function name.
Please make me know if you would like me to abbreviate.

**Comment by igaw** [comment]
Generally, it's simpler to review stuff if a patch is only doing one thing at a time. I am not enforcing this for plugins (yet).

About naming scheme, the best strategy to avoid namespace clashes is to use `static` when ever possible and for the rest a short prefix should do the trick; having 8 chars prefix is a bit excessive IMO. Just saying, you maintain the plugin, it's up to you. I am just here to complain at random times :)

**Pull request merged**